google-ads-migration-examples/src/main/java/com.google.ads.googleads.migration/createcampaign/CreateCompleteCampaignAdWordsApiOnly.java

```diff
@@ -0,0 +1,429 @@
+
+      // Optional: set the status.
+      expandedTextAdGroupAd.setStatus(AdGroupAdStatus.PAUSED);
+
+      // Create the operation.
+      AdGroupAdOperation adGroupAdOperation = new AdGroupAdOperation();
+      adGroupAdOperation.setOperand(expandedTextAdGroupAd);
+      adGroupAdOperation.setOperator(Operator.ADD);
+
+      operations.add(adGroupAdOperation);
+    }
+
+    // Add the ads.
+    AdGroupAdReturnValue result =
+      adGroupAdService.mutate(operations.toArray(new AdGroupAdOperation[operations.size()]));
+
+    // Display the ads.
+    Arrays.stream(result.getValue())
+      .map(adGroupAdResult -> (ExpandedTextAd) adGroupAdResult.getAd())
+      .forEach(
+        newAd ->
+          System.out.printf("Expanded text ad with ID %d" +
+              "and headline '%s - %s' was found in ad group with ID %d.%n",
+            newAd.getId(),
+            newAd.getHeadlinePart1(),
+            newAd.getHeadlinePart2(),
+            adGroup.getId()));
+
+    return result.getValue();
+  }
+
+  /**
+   * Creates an ad group.
+   *
+   * @param adWordsServices the Google AdWords services interface.
+   * @param session the client session.
+   * @param campaign the campaign for the ad group.
+   * @throws RemoteException if the API request failed due to other errors.
+   */
+  private AdGroup createAdGroup(AdWordsServicesInterface adWordsServices,
+                                AdWordsSession session, Campaign campaign)
+    throws RemoteException {
+    // Get the AdGroupService.
+    AdGroupServiceInterface adGroupService =
+      adWordsServices.get(session, AdGroupServiceInterface.class);
+
+    // Create ad group.
+    AdGroup adGroup = new AdGroup();
+    adGroup.setName("Earth to Mars Cruises #" + System.currentTimeMillis());
+    adGroup.setStatus(AdGroupStatus.ENABLED);
+    adGroup.setCampaignId(campaign.getId());
+
+    // Set the rotation mode.
+    AdGroupAdRotationMode rotationMode = new AdGroupAdRotationMode(AdRotationMode.OPTIMIZE);
+    adGroup.setAdGroupAdRotationMode(rotationMode);
+
+    // Create ad group bid.
+    BiddingStrategyConfiguration biddingStrategyConfiguration = new BiddingStrategyConfiguration();
+    Money cpcBidMoney = new Money();
+    cpcBidMoney.setMicroAmount(10_000_000L);
+    CpcBid bid = new CpcBid();
+    bid.setBid(cpcBidMoney);
+    biddingStrategyConfiguration.setBids(new Bids[] {bid});
+    adGroup.setBiddingStrategyConfiguration(biddingStrategyConfiguration);
+
+    // Create the operation.
+    AdGroupOperation operation = new AdGroupOperation();
+    operation.setOperand(adGroup);
+    operation.setOperator(Operator.ADD);
+
+    AdGroupOperation[] operations = new AdGroupOperation[] {operation};
+
+    // Add the ad group.
+    AdGroupReturnValue result = adGroupService.mutate(operations);
+
+    AdGroup adGroupResult = result.getValue(0);
+    // Display the new ad group.
+    System.out.printf("Ad group with ID '%d' and name %s was created.%n",
```

Comment:
When printing, please do not quote IDs but quote strings, so `"Ad group with ID %d and name '%s' was created.%n"` here.

Reply:
Done.